roles/photoprism/tasks/main.yml

```diff
@@ -0,0 +1,32 @@
+---
+- name: Create config directory
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: 0777
+  with_items:
+    - "{{ photoprism_data_directory }}/storage"
+
+- name: Install Docker container
+  docker_container:
+    name: photoprism
+    image: photoprism/photoprism
+    pull: true
+    ports:
+      - "{{ photoprism_port }}:2342"
+    volumes:
+      - "/etc/localtime:/etc/localtime:ro"
+      - "{{ photoprism_data_directory }}/storage:/photoprism/storage:rw"
+      - "{{ photoprism_photos_directory }}:/photoprism/originals:rw"
+    env:
+      PHOTOPRISM_SITE_URL: "https://photoprism.{{ ansible_nas_domain }}"
+      PHOTOPRISM_ADMIN_PASSWORD: "insecure"
+    restart_policy: unless-stopped
+    memory: 4g
+    labels:
+      traefik.enable: "{{ photoprism_available_externally  }}"
+      traefik.http.routers.gitlab.rule: "Host(`photoprism.{{ ansible_nas_domain }}`)"
```

Comment:
I'd also like to have the subdomain part of the Host traefik rule configurable instead of hard coded to `photoprism`